torch_xla/torch_xla.py

```diff
@@ -101,19 +101,21 @@ def foo2(x):
   """
 
   @contextlib.contextmanager
-  def _step():
+  def _compile():
     saved_eager_mode_status = torch_xla._XLAC._get_use_eager_mode()
     torch_xla._XLAC._set_use_eager_mode(False)
     # Clear pending operations
-    sync()
+    xm.mark_step()
```

Comment:
Why is this changed to `mark_step`?

Reply:
there are two `sync` in the `torch_xla.compile`, one before traicng and one after tracing. In the C++ layer I am just looking at the python stack trace and it is not easy to figure out which one is which. I think I have 3 options

1. wrap one of the `sync` in to a function like `sync_graph_before_compile`
2. look at line number
3. use `mark_step` for one and `sync` for the other

I don't like first two so I ended up using the last one.

Reply:
Hahaha that's hacky

(1) is definitely the way to go here just to make clear what you're trying to do, and make this output more immediately readable to anyone else looking at the stack trace; there's not anything about `mark_step` that clearly communicates "before". You can add inner functions to avoid polluting the global namespace:

```
def compile

  def sync_before(): # or more explicitly, `clear_pending_ops`
      sync()

  def sync_after():
     sync()
```

Regarding the depth of the stack trace, we can easily flatten it a level by having `sync` just call `torch_xla._XLAC._xla_step_marker` directly instead of calling mark_step.